src/app/store/user/user.effects.ts

```diff
@@ -1,94 +1,123 @@
+import * as userActions from './user.actions'
+import { UsersService } from '../../service/users/users.service'
 
 @Injectable()
-
 export class UserEffects {
-
-    constructor(private actions$: Actions, private router: Router, private service: UsersService) { }
-    loadUsers$ = createEffect(() => this.actions$.pipe(
-        ofType(userActions.loadUsers),
-        mergeMap(() => this.service.getAllUsers().pipe(
-            map(users => userActions.loadUsersSuccess({ users })),
-            catchError(error => of(userActions.loadUsersFailure({ error })))
+  constructor(
+    private actions$: Actions,
+    private router: Router,
+    private service: UsersService
+  ) {}
+  loadUsers$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(userActions.loadUsers),
+      mergeMap(() =>
+        this.service.getAllUsers().pipe(
+          map((users) => userActions.loadUsersSuccess({ users })),
+          catchError((error) => of(userActions.loadUsersFailure({ error })))
         )
+      )
+    )
+  )
+
+  signInUser$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(userActions.signInUser),
+      switchMap(({ body }) =>
+        this.service.signInUser(body).pipe(
+          map((response) =>
+            userActions.signInUserSuccess({
+              currentUser: response.user,
+              isLoggedIn: response.token,
+              userId: response.id,
+            })
+          ),
+          tap(({ userId, isLoggedIn }) => {
+            localStorage.setItem('id', userId)
+            localStorage.setItem('isLoggedIn', isLoggedIn)
+            this.router.navigate(['/tasks-list'])
+          }),
+          catchError((error) => of(userActions.signInUserFailure(error)))
         )
-    ))
+      )
+    )
+  )
 
-    signInUser$ = createEffect(() => this.actions$.pipe(
-        ofType(userActions.signInUser),
-        switchMap(({ body }) =>
-            this.service.signInUser(body).pipe(
-                map((response) =>
-                    userActions.signInUserSuccess({
-                        currentUser: response.user,
-                        isLoggedIn: response.token,
-                        userId: response.id
-                    })
-                ),
-                tap(({ userId, isLoggedIn }) => {
-                    localStorage.setItem("id", userId)
-                    localStorage.setItem("isLoggedIn", isLoggedIn)
-                    this.router.navigate(["/tasks-list"])
-                }),
-                catchError((error) =>
-                    of(userActions.signInUserFailure(error))
-                )
+  getCurrentUser$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(userActions.getCurrentUser),
+      switchMap(() =>
+        this.service.getUser().pipe(
+          map((user) =>
+            userActions.getCurrentUserSuccess({
+              currentUser: user,
+              isLoggedIn: localStorage.getItem('isLoggedIn'),
```

Comment:
if you get a current user shouldn't isLoggedIn be true anyway? if the user is not logged in you won't be able to get any data and if he is logged in then the data of course will be fetched, review this portion